### DISCUSSION TITLE
useGroupBy; not putting grouped column as first column

### DISCUSSION THREAD

**Discussion by richardhealy** [TanStack/table#2291] [General]
In the Codesandbox [demo ](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/groupin) for useGroupBy, it seems that the grouped column is always moved to the first column. Is there a way to prevent this? Or even sort of column order override that can be used?

**Comment by vancassa** [comment]
Is there any update on this? 🤔

**Comment by aaronlna** [comment]
I ran into this issue when wanting a selection column on the leftmost side of my table while supporting grouping using the `useGroupBy` hook.

I was able to work around this by setting `display: flex` on my table-rows and setting `order: -1` on my selection data-cells manually. This may work for you depending on the use-case. It seems like the `useColumnOrder` hook _should_ support the ability to sort grouped columns. It's not yet clear to me why this isn't supported.
